source/server/overload_manager_impl.cc

```diff
@@ -374,6 +399,48 @@ bool OverloadManagerImpl::registerForAction(const std::string& action,
   return true;
 }
 
+bool OverloadManagerImpl::tryAllocateResource(OverloadReactiveResourceName resource_name,
+                                              uint64_t increment,
+                                              ReactiveResourceUpdateCallbacks& cb) {
+  const auto reactive_resource = reactive_resources_.find(resource_name);
+  if (reactive_resource != reactive_resources_.end()) {
+    if (reactive_resource->second.tryAllocateResource(increment)) {
+      cb.dispatcher().post(
+          [&cb, resource_usage = reactive_resource->second.currentResourceUsage()]() {
+            cb.onSuccess(resource_usage);
+          });
+      return true;
+    } else {
+      cb.dispatcher().post([&cb]() { cb.onFailure(); });
+      return true;
+    }
+  } else {
+    ENVOY_LOG(warn, " {Failed to allocate unknown reactive resource }");
+    return false;
```

Comment:
Should this branch call `cb.dispatcher().post([&cb]() { cb.onFailure(); });`

if so, I think the return value of this method could be removed.